### PR TITLE
Support for node >15 and npm>7

### PR DESCRIPTION
**Pull request by tlxzilia**
Fixes https://github.com/nix-community/npmlock2nix/issues/153

### CODE REVIEW

File: internal.nix [hunk 1]
```diff
@@ -294,14 +294,20 @@ rec {
         else version);
       dependencies = if (content ? dependencies) then lib.mapAttrs patchDep content.dependencies else { };
       devDependencies = if (content ? devDependencies) then lib.mapAttrs patchDep content.devDependencies else { };
+      defaultScripts = if (content ? scripts) then content.scripts else { };
+      scripts = defaultScripts // {
+        preinstall = "${preinstallScript};" + (if (defaultScripts ? preinstall) then defaultScripts.preinstall else "");
```
Comment (andir) [inline internal.nix:299]:
```suggestion
        preinstall = "${preinstallScript};" + (lib.optionalString (defaultScripts ? preinstall) defaultScripts.preinstall);
```

**Review by andir — COMMENTED** [review]
Just a quick pass. Thanks for the contribution!

With this change we will not be able to use an `npm` below version 15 anymore (for non-trivial) projects, is that right? I don't mind dropping support for versions that are out of support. Is there still some LTS support for an older version from upstream node?

The way you implemented hooks seems like they will only fire for the top-level package and not for any of the dependencies any more. Is that correct? If not I don't get how patching a single `package.json` file would propagate to all the dependencies.

File: internal.nix (hunk 1 above)
Comment (tlxzilia) [reply internal.nix:299]: Ahh, I will need to think about this a little bit more. Worked for my workflow but still had to override single package with patchesShebangs.

**Comment by tlxzilia** [comment]
Reseted my fork because of errors and this got closed. Will reopen another PR soon.

**Pull request merged**